sdk/identity/azure-identity/tests/test_imds_credential.py

```diff
@@ -176,6 +177,45 @@ def test_identity_config():
     assert token == expected_token
 
 
+def test_imds_url_override():
+    url = "https://localhost/token"
+    expected_token = "***"
+    scope = "scope"
+    now = int(time.time())
+
+    transport = validating_transport(
+        requests=[
+            Request(base_url=url),
```

Comment:
Would a request without required headers get sent in this case, since we assume that the endpoint is available?

Reply:
It does in this case because the test is wrong, thanks for pointing that out! 🤣